src/g_array.c

```diff
@@ -516,46 +536,34 @@ void garray_arrayviewlist_fillpage(t_garray *x,
     topItem = (int)fTopItem;
     if (!a)
     {
-        /* FIXME */
+            /* FIXME */
         pd_error(0, "error in garray_arrayviewlist_new()");
+        return;
     }
 
-    if (page < 0) {
-      page = 0;
-      sys_vgui("pdtk_array_listview_setpage %s %d\n",
-               x->x_realname->s_name,
-               (int)page);
-    }
-    else if ((page * ARRAYPAGESIZE) >= a->a_n) {
-      page = (int)(((int)a->a_n - 1)/ (int)ARRAYPAGESIZE);
-      sys_vgui("pdtk_array_listview_setpage %s %d\n",
-               x->x_realname->s_name,
-               (int)page);
-    }
-    sys_vgui(".%sArrayWindow.lb delete 0 %d\n",
-             x->x_realname->s_name,
-             ARRAYPAGESIZE - 1);
+    page = garray_arrayviewlist_setpage(x->x_realname, a, page, ARRAYPAGESIZE);
+
+    sys_vgui("::dialog_array::listview_setdata {%s} %ld",
+        x->x_realname->s_name,
+        (long long)(page * ARRAYPAGESIZE));
     for (i = page * ARRAYPAGESIZE;
          (i < (page + 1) * ARRAYPAGESIZE && i < a->a_n);
          i++)
     {
         yval = *(t_float *)(a->a_vec + \
                elemsize * i + yonset);
```

Comment:
see `garray_arrayviewlist_new`

Reply:
also fixed (by dropping the duplicate implementation )